packages/openapi-v3-types/src/openapi-v3-spec-types.ts

```diff
@@ -0,0 +1,395 @@
+ * }
+ * ```
+ */
+export interface ContentObject {
+  [mediatype: string]: MediaTypeObject;
+}
+
+/**
+ * Each Media Type Object provides schema and examples for the media type
+ * identified by its key.
+ * <p>Specification:
+ * https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#media-type-object
+ */
+export interface MediaTypeObject extends ISpecificationExtension {
+  schema?: SchemaObject | ReferenceObject;
+  examples?: [ExampleObject | ReferenceObject];
+  example?: ExampleObject | ReferenceObject;
+  encoding?: EncodingObject;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `ParameterObject`
+ */
+export interface EncodingObject extends ISpecificationExtension {
+  // [property: string]: EncodingPropertyObject;
+  [property: string]: EncodingPropertyObject | ExtensionValue; // Hack for allowing ISpecificationExtension
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `ParameterObject`
+ */
+export interface EncodingPropertyObject {
+  contentType?: string;
+  headers?: {[key: string]: HeaderObject | ReferenceObject};
+  style?: string;
+  explode?: boolean;
+  allowReserved?: boolean;
+  [key: string]: ExtensionValue; // (any) = Hack for allowing ISpecificationExtension
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `ParameterObject`
+ */
+export interface HeaderObject extends ParameterObject {}
+
+/**
+ * Describes a single operation parameter.
+ * A unique parameter is defined by a combination of a name and location.
+ * <p>Specification
+ * https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#parameter-object
+ */
+export interface ParameterObject extends ISpecificationExtension {
+  name: string;
+  in: ParameterLocation;
+  description?: string;
+  required?: boolean;
+  deprecated?: boolean;
+  allowEmptyValue?: boolean;
+
+  style?: ParameterStyle;
+  explode?: boolean;
+  allowReserved?: boolean;
+  schema?: SchemaObject | ReferenceObject;
+  examples?: {[param: string]: ExampleObject | ReferenceObject};
+  example?: ExtensionValue;
+  content?: ContentObject;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `ParameterObject`
+ */
+export interface ExampleObject {
+  summary?: string;
+  description?: string;
+  value?: ExtensionValue;
+  externalValue?: string;
+  [property: string]: ExtensionValue; // Hack for allowing ISpecificationExtension
+}
+
+export interface ReferenceObject {
+  $ref: string;
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `SchemaObject`
+ */
+export interface DiscriminatorObject {
+  propertyName: string;
+  mapping?: {[key: string]: string};
+}
+
+/**
+ * Describes an encoding object, copied from 'openapi3-ts' due to overriding
+ * its `SchemaObject`
+ */
+export interface XmlObject extends ISpecificationExtension {
```

Comment:
Should we call it `XMLObject` instead of `XmlObject`? ... I've only ever seen `XML` in all caps.

Reply:
copied it from the community module didn't notice it's camel case, according to the document yes I think it should be `XML` https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#xml-object